File: app/src/main/java/com/example/ndk/player/MusicPlayer.java
package com.example.ndk.player;


/**
 * @author yudongliang
 * create time 2021-08-11
 * describe :
 */
public class MusicPlayer {

    static {
        System.loadLibrary("music-player");
    }

    /**
     * ffmpeg 版本号
     */
    public native String ffmpegVersion();

    /**
     * 得到音频 meta 信息
     */
    public native void mediaInfo(String url);


}

File: app/src/main/java/com/example/ndk/player/MusicPlayer.kt
package com.example.ndk.player

/**
 * @author yudongliang
 * create time 2021-08-10
 * describe : 音乐播放器
 */
class MusicPlayer {

    //开始播放音频
    external fun play(url:String): String

    companion object {

        init {
            System.loadLibrary("music-player")
        }
    }

}